### ISSUE TITLE
EventCallback not working for generic components

### ISSUE THREAD

**Issue by stsrki** [aspnet/AspNetCore#12627]
This issue is continuation of #8385 

The original problem is fixed but now the problem is that generic **typeparam** must always be defined when using custom component. It seems that Blazor is not recognizing it implicitly.

The minimal sample can be found here: https://github.com/stsrki/BlazorGenericEventIssue

### Not working:

This will giving me error because **TValue** is not defined:

`<SelectEdit Value="@someValue" ValueChanged="@((v) => someValue=v)">`

> Error	CS0029	Cannot implicitly convert type 'Microsoft.AspNetCore.Components.UIWheelEventArgs' to 'int'
> Error	CS1662	Cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type
> Error	CS0029	Cannot implicitly convert type 'int' to 'System.Threading.Tasks.Task'
> Error	CS0266	Cannot implicitly convert type 'object' to 'int'. An explicit conversion exists (are you missing a cast?)
> 

### Working:

This is working only after **TValue** is set. It is not recognized from the context.

`<SelectEdit TValue="int" Value="@someValue"ValueChanged="@((v) => someValue=v)">`

### Expected behavior:

`TValue` should be automatically detected from the supplied value(s).

**Comment by marin-bratanov** [comment]
May be related/duplicate of [https://github.com/aspnet/AspNetCore/issues/12226](https://github.com/aspnet/AspNetCore/issues/12226)

**Comment by pranavkm** [comment]
As @marin-bratanov this is pretty similar to #12226. We'll use the other issue to track further work.
